pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -114,6 +120,11 @@ func (c *clusterMemberRemovalController) sync(ctx context.Context, syncCtx facto
 // attemptToScaleDown attempts to remove a voting member only once we have identified that
 // a Machine resource is being deleted and a replacement member has been created
 func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context, recorder events.Recorder) error {
+	scalingStrategy, err := ceohelpers.GetBootstrapScalingStrategy(c.operatorClient, c.namespaceLister, c.infrastructureLister)
```

Comment:
I think I have something simpler for you, I would check two things in sync() already:
* is it currently bootstrapping?
* is it safe based on the number of healthy/unhealthy members?

Right next to the method you've moved into common, there's another called `CheckSafeToScaleCluster` which does exactly that. I think you can just call it early in `sync` and just make it a little smarter, for example by updating `IsBootstrapComplete` to also include an etcdclient memberList to see if there's still a bootstrap node in there.

Reply:
done 👍🏽

Reply:
not quite what I meant, but thanks for adding this one here too.

Reply:
what am i missing ?? .. i still can change it

Reply:
all good, you fixed this with the other comment already!